### PR TITLE
Windows libswami adaptation

### PR DESCRIPTION
**Pull request by jjceresa**
This is the start of changes to get `libswami target` configured using CMake and compiled on Windows without errors.
Configuration (CMake), is dependent of actual `swami-libinstpatch` build and installation (shared library, includes files, libinstpatch.pc).

### CODE REVIEW

File: src/libswami/SwamiLog.h [hunk 1]
```diff
@@ -70,10 +70,10 @@ extern void _swami_pretty_log_handler (GLogLevelFlags flags,
 	 	 	     __FILE__, __PRETTY_FUNCTION__, __LINE__, \
 	 	 	     format, ## args);
 #else
-#define SWAMI_DEBUG(format, args...)
+#define SWAMI_DEBUG(format, args)
 #endif
 
-#define SWAMI_INFO(format, args...) \
+#define SWAMI_INFO(format, args) \
```
Comment (derselbst) [inline src/libswami/SwamiLog.h]: Variadic macros are a C99 feature. You need VS2015 to use it. Your change breaks compilation I'm afraid. You may convert these macros to normal functions though, to make it work.

**Review by derselbst — COMMENTED** [review]
Ok, one note.

**Comment by jjceresa** [comment]
>Variadic macros are a C99 feature. You need VS2015 to use it. Your change breaks compilation I'm afraid. You may convert these macros to normal functions though, to make it work.

These changes, aren't correct, sorry. I will see how to correct this using normal functions.

**Comment by jjceresa** [comment]
Note that for Windows:  SWAMI_DEBUG(format, args...),SWAMI_INFO(format, args...),SWAMI_PARAM_ERROR(param), SWAMI_CRITICAL(format, args...) are not implemented because they aren't called in any others component of swami application. For now, these aren't important.
I'm fine with this PR, and consider that ready.

**Comment by jjceresa** [comment]
PR `Windows libswami adaptation` is ready:
- CMake (configure and generate) is OK.
- Build is OK with 0 errors and 0 warning.
It could be merged if you agree.

Next PRs would be:
- PR `Windows libswamigui + swami adaptation` then,
- PR `Windows plugin tune adaptation' then
- PR `Windows plugin fluidsynth adaptation' .

**Comment by jjceresa** [comment]
This PR is ready,
Next PRs would be.
- `Windows libswamigui-swami adaptation` then,
- `Windows tune-plugin adaptation` then,
- `Windows fluidsynth-plugin adaptation`.

**Pull request merged**